http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/MockSamDAO.scala

```diff
@@ -75,7 +82,8 @@ class MockSamDAO extends SamDAO[IO] {
         IO(runtimes += (r, userEmailToAuthorization(creatorEmail)) -> runtimeActions)
       case r: PersistentDiskSamResource =>
         IO(persistentDisks += (r, userEmailToAuthorization(creatorEmail)) -> diskActions)
-      case _ => IO(throw new Exception("Invalid resource to create"))
+      case r: AppSamResource => IO.unit //TODO
+      case _                 => IO(throw new Exception("Invalid resource to create"))
```

Comment:
let's not put `_` here, we lose the benefit of using `sealed` this way, because we do want to get warnings if we don't exhaust the cases properly